ci-automation/garbage_collect_releases.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+#
+# Copyright (c) 2024 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# >>> This file is supposed to be SOURCED from the repository ROOT. <<<
+#
+# garbage_collect_releases() should be called after sourcing.
+#
+#  OPTIONAL INPUT
+#  - Number releases to keep per channel. Defaults to 10.
+#  - Number of LTS channels to keep. Defaults to 2 (i.e. the current and the previous (deprecated) LTS).
+#  - DRY_RUN (Env variable). Set to "y" to just list what would be done but not
+#            actually purge anything.
+
+# Flatcar build cache releases artifacts garbage collector.
+#  This script removes release artifacts of past releases from the build cache.
+#  Note that release artifacts are copied to official mirrors upon release, so there's
+#   no need to keep a copy on the build cache server.
+
+function garbage_collect_releases() {
+    # Run a subshell, so the traps, environment changes and global
+    # variables are not spilled into the caller.
+    (
+        set -euo pipefail
+
+        _garbage_collect_releases_impl "${@}"
+    )
+}
+# --
+
+function _garbage_collect_releases_impl() {
+    local keep_per_chan="${1:-10}"
+    local keep_lts_releases="${2:-2}"
+    local dry_run="${DRY_RUN:-}"
+
+    echo
+    echo "Number of versions to keep per channel: '${keep_per_chan}'"
+    echo "Number of LTS major releases to keep: '${keep_lts_releases}'"
+    echo
+
+    source ci-automation/ci_automation_common.sh
+    local sshcmd="$(gen_sshcmd)"
+
+    local keep="$( 
```

Comment:
Quotes in assignment are not necessary.
```suggestion
    local keep=$( 
```